src/framework/emqttb_worker.erl

```diff
@@ -78,19 +78,20 @@ init_per_group(Module, GroupID, Opts) ->
 %% Getters/utilities
 %%--------------------------------------------------------------------------------
 
--spec send_after(counters:counters_ref() | non_neg_integer(), _Message) -> reference().
-send_after(I, Message) when is_integer(I) ->
-  erlang:send_after(I, self(), Message);
-send_after(CRef, Message) ->
-  I = counters:get(CRef, 1),
+-spec repeat(non_neg_integer(), fun(() -> _)) -> ok.
+repeat(0, _) ->
+  ok;
+repeat(N, Fun) ->
+  Fun(),
+  repeat(N - 1, Fun).
+
+-spec send_after(non_neg_integer(), _Message) -> reference().
+send_after(I, Message) ->
   erlang:send_after(I, self(), Message).
 
--spec send_after_rand(counters:counters_ref() | non_neg_integer(), _Message) -> reference().
+-spec send_after_rand(non_neg_integer(), _Message) -> reference().
 send_after_rand(I, Message) when is_integer(I) ->
-  erlang:send_after(rand:uniform(I + 1), self(), Message);
-send_after_rand(CRef, Message) ->
-  I = counters:get(CRef, 1),
```

Comment:
I think we still need this clause, since this fn is imported in some behaviors.

Reply:
I tried to change ever occurance. Unless I missed something, there is no promise to keep a stable API :)

Reply:
True, I think that's all occurrences.  Ignore me.  :speak_no_evil: